### PR TITLE
EOS-8749: Revert global indexes on create bucket error

### PR DESCRIPTION
**Pull request by dsurnin**
Signed-off-by: Dmitrii Surnin <dmitrii.surnin@seagate.com>

### CODE REVIEW

**Comment by dsurnin** [comment]
http://eos-jenkins.colo.seagate.com/job/S3server/job/s3-github-test/89/

**Comment by johnbent** [comment]
@dsurnin , are you getting the reviews you need on this?

File: server/s3_bucket_metadata_v1.cc [hunk 1]
```diff
@@ -482,12 +469,17 @@ void S3BucketMetadataV1::save_replica() {
...
+  auto kv_state = motr_kv_writer->get_state();
+  if (!should_cleanup_global_idx) {
+    if (kv_state == S3MotrKVSWriterOpState::failed_to_launch) {
+      state = S3BucketMetadataState::failed_to_launch;
+    } else {
+      state = S3BucketMetadataState::failed;
+    }
+    this->handler_on_failed();
   } else {
-    state = S3BucketMetadataState::failed;
```
Comment (bkirunge7) [inline server/s3_bucket_metadata_v1.cc:488]: will this state value will be used at other places

**Review by evgeniy-brazhnikov — APPROVED** [review]
Good for me

File: server/s3_bucket_metadata_v1.cc (hunk 1 above)
Comment (dsurnin) [reply server/s3_bucket_metadata_v1.cc:488]: yes, this is an operation status

**Review by bkirunge7 — COMMENTED** [review]
@dsurnin see if you can add a System test for this use KV tool check if index is exists

**Comment by dsurnin** [comment]
> 
> 
> @dsurnin see if you can add a System test for this use KV tool check if index is exists

I already modified existing tests and added a couple of new

**Comment by bkirunge7** [comment]
> > @dsurnin see if you can add a System test for this use KV tool check if index is exists
> 
> I already modified existing tests and added a couple of new

I see unit tests added, no system tests, like create a bucket using aws cli with fault injection enabled, and see if it does cleanup in index table.  If its feasible then fine.

**Comment by dsurnin** [comment]
> 
> 
> > > @dsurnin see if you can add a System test for this use KV tool check if index is exists
> > 
> > 
> > I already modified existing tests and added a couple of new
> 
> I see unit tests added, no system tests, like create a bucket using aws cli with fault injection enabled, and see if it does cleanup in index table. If its feasible then fine.

I didn't bring any new functionality, just reused old one, already proved and tested. In unit tests I checked that it is properly called. I think it should be enough.
Approach you suggest will require several server restarts - we do not use FI bucket creation now in tests, so stop prev, reconfigure startup, run single test for this particular scenario, stop and reconfigure as it was previous,start and continue testing. It will increase build time and test logic since we do not assume any restarting now.

**Review by bkirunge7 — APPROVED** [review]

**Pull request merged**